### ISSUE TITLE
Feature Request: Support EC2 Optimize CPU Options

### ISSUE THREAD

**Issue by foax** [terraform-providers/terraform-provider-aws#4490]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS have announced the availability of Optimize CPUs for Amazon EC2 instances. This provides the ability to launch EC2 instances with a specific number of CPU cores, and to specify the number of threads per CPU core (effectively enabling or disabling Hyperthreading).

These options can only be applied on EC2 instance launch.

The main benefit of this capability is to allow CPUs to be masked to comply with software licensing requirements.

Note that each EC2 instance type has a set of valid configurations that can be applied. See https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-optimize-cpu.html#cpu-options-supported-instances-values for a table of these options.

### New or Affected Resource(s)

* aws_instance

### Potential Terraform Configuration

```hcl
resource "aws_instance" "example" {
  ami                  = "${data.aws_ami.ubuntu.id}"
  instance_type        = "r4.4xlarge"
  cpu_cores            = "4"
  cpu_threads_per_core = "2"
}
```

### References

* Amazon release announcement: https://aws.amazon.com/about-aws/whats-new/2018/05/introducing-optimize-cpus-for-amazon-ec2-instances/
* AWS EC2 documentation: https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instance-optimize-cpu.html

**Comment by smastrorocco** [comment]
Has there been any movement on this feature?

**Comment by bflad** [comment]
This has been released in [version 1.29.0 of the AWS provider](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md#1290-july-26-2018). Please see the [Terraform documentation on provider versioning](https://www.terraform.io/docs/configuration/providers.html#provider-versions) or reach out if you need any assistance upgrading.
